source/SharedCrtResourceManager.cpp

```diff
@@ -136,6 +136,20 @@ int SharedCrtResourceManager::buildClient(const PlainConfig &config)
     return SharedCrtResourceManager::SUCCESS;
 }
 
+void SharedCrtResourceManager::initializeAWSHttpLib()
+{
+    if (!initialized)
+    {
+        LOG_WARN(TAG, "Tried to aws_http_library_init but the SharedCrtResourceManager has not yet been initialized!");
+    }
+    if (initializedAWSHttpLib)
+    {
+        LOG_WARN(TAG, "Tried to aws_http_library_init but it was already initialized!");
+    }
+    aws_http_library_init(getAllocator());
+    initializedAWSHttpLib = true;
```

Comment:
I think you actually need a lock for this value, or make it atomic. I can imagine a situation where you have multiple features initializing at the same time and you're reading/writing to this value concurrently. Likely have a race condition here